codegen/gateway.go

```diff
@@ -236,14 +236,14 @@ func NewCustomClientSpec(
 		if _, ok := clientConfig.Config[f]; !ok {
 			return nil, errors.Errorf(
 				"client config %q must have %q field for type custom",
-				instance.JSONFileName,
+				instance.YAMLFileName,
 				f,
 			)
 		}
 	}
 
 	clientSpec := &ClientSpec{
-		JSONFile:           instance.JSONFileName,
```

Comment:
There is existing code that relies on `JSONFile` property. Can you keep the existing property and add `YAMLFile` as a new property?